### ISSUE TITLE
when cluster members click on scale down the status code should be 403 instead of 500 status code

### ISSUE THREAD

**Issue by anupama2501** [rancher/rancher#34253]
**Rancher Server Setup**
- Rancher version: v2.6-head 4def656
- Installation option (Docker install/Helm Chart): Docker install
   - If Helm Chart, Kubernetes Info: 
         Cluster Type (RKE1, RKE2, k3s, EKS, etc):
         Node Setup:
         Version:
- Proxy/Cert Details: Self-signed

**Information about the Cluster**
- Kubernetes version: v1.21.3
- Cluster Type (Local/Downstream): Downstream rke1 node driver

**Describe the bug**
When cluster members try to scale down or scale up a node pool, status code displayed is 500 with a `403` response.

**To Reproduce**
1. Enable auth in the cluster
2. Create an rke1 node driver cluster with 3 worker nodes
3. Add a member of auth provider in the cluster as a cluster member from RBAC --> members
4. login as the cluster member
5. From cluster management page --> click on scale down

**Result**
The scale down results in `500` status code with a response code `message: "Error updating node c-qdk5r:m-ddwgn by scaledown : Forbidden 403: nodes.management.cattle.io \"m-ddwgn\" is forbidden: User \"u-ckrl4grxg5\" cannot update resource \"nodes\" in API group \"management.cattle.io\" in the namespace \"c-qdk5r\""
`

**Expected Result**
Status code should be 403 as well. 

**Screenshots**
<img width="400" alt="Screen Shot 2021-08-13 at 4 48 34 PM" src="https://user-images.githubusercontent.com/38144301/129427718-c4f8cca1-f474-4dc3-aea1-ab942f2efab1.png">

**Comment by aiyengar2** [comment]
Since the original linked commit https://github.com/rancher/rancher/commit/4def6561042e4603cd652ac0a3bdb829069f1dbd is contained within `v2.6.0-rc3` and the fix in https://github.com/rancher/rancher/commit/d1b016d4c11684842ef417cf83542e0027096fe9 is not contained within an RC, I will use the following branches to reproduce and test the fix:

Reproduce: `v2.6.0-rc3`
Test Fix: `v2.6-head (2778ecabaeae72be86bd90a4f6127d72c1a9b686)`

#### External resources used for reproducing

Prior to following these steps, ensure that you have two separate accounts that you can use to test this.

I used two GitHub accounts with the GitHub OAuth Provider
- USER1 is the admin user and USER2 is the cluster member. 
- WINDOW1 is the browser / incognito window used by USER1 and WINDOW2 is the browser / incognito window used by USER2

#### Steps to Reproduce


- Provision a node and install Docker on it via `curl https://releases.rancher.com/install-docker/20.10.sh | sh`
- Install Rancher on the node

```
# RANCHER_VERSION=v2.6.0-rc3
RANCHER_VERSION=v2.6-head

docker run -d --restart=unless-stopped   -p 80:80 -p 443:443   --privileged   rancher/rancher:${RANCHER_VERSION}
```

###### In WINDOW1:
- Login to Github on the browser that you open up Rancher in using USER1's account
- Login to Rancher as the admin user
- Navigate to Users & Authentication -> Auth Provider -> Google
- Follow the instructions to set up the GitHub OAuth Provider
- Navigate to Cluster Management -> Create. Ensure RKE1 is enabled and set up a Digital Ocean cluster with 1 etcd, 1 controlplane, and 3 workers

###### In WINDOW2:
- While the cluster is provisioning, log in using a separate browser or an incognito window to USER2's Github account
- In that separate browser, open Rancher and click on `Log In with Github`

###### In WINDOW1:
- Once the cluster is done provisioning, go to Explore Cluster -> <cluster-name> -> RBAC -> Cluster Members -> Add
- Under `Select Member`, type in USER2's Github Account. Ensure that Cluster Permissions has selected `Member` and click `Create`
- Once done, you should see two users: `Default Admin` (USER1) and USER2

###### In WINDOW2:
- Refresh the page
- Click on Cluster Management -> <cluster-name>

#### 2.6.0-rc3

Scale Down option is observed

![arvind-rancher-260rc3 cp-dev rancher space_dashboard_c_c-hsw5m_manager_provisioning cattle io cluster](https://user-images.githubusercontent.com/13127443/130143321-3ffd3377-0a8b-4556-a34d-7f2bda5389d6.png)

#### 2.6-head

Scale Down option is **not** observed, which seems to validate the fix Darren put in worked.

However, one of the two errors keeps popping up:

1. On initially landing on the page, we see `{ "message": "Cannot read property 'startsWith' of undefined", "statusCode": 500 }`

![arvind-rancher-26head cp-dev rancher space_dashboard_c_c-7nlx8_manager_provisioning cattle io cluster](https://user-images.githubusercontent.com/13127443/130143332-c4c5d202-e752-47a0-914b-b995e5849811.png)

2. On refreshing the page, a call to `/v3/clusterregistrationtokens` is made and we see `{ "baseType": "error", "code": "Forbidden", "message": "clusterregistrationtokens.management.cattle.io is forbidden: User \"u-vgjfy5lxnf\" cannot create resource \"clusterregistrationtokens\" in API group \"management.cattle.io\" in the namespace \"c-7nlx8\"", "status": 403, "type": "error", "statusCode": 403 }`

![arvind-rancher-26head cp-dev rancher space_dashboard_c_c-7nlx8_manager_provisioning cattle io cluster_fleet-default_c-7nlx8](https://user-images.githubusercontent.com/13127443/130143631-006f83d3-1a9f-4ae5-b59e-4e93166d3a6a.png)

For the second issue, no network calls are made (except to get the favicon.png) and the following text is logged in the console:

```
actions.js:270 Find: [management] provisioning.cattle.io.cluster fleet-default/c-7nlx8
resource-instance.js:516 Starting wait for set provisioner
resource-instance.js:529 Wait for set provisioner done immediately
actions.js:157 Find All: [management] management.cattle.io.nodetemplate
resource-instance.js:542 Wait for set provisioner done
```

---

cc: @sowmyav27 @vincent99 I think that the above two issues do not seem to be backend related, right? From what I can tell, the 500 error is probably coming from the UI and the 403 should be hidden by the UI (or not requested in the first place).

I'm not sure whether I should open up a separate issue / UI issues for this, so please let me know!

**Comment by aiyengar2** [comment]
403 issue is tracked in https://github.com/rancher/dashboard/issues/3852.

500 issue is tracked in https://github.com/rancher/dashboard/issues/3942.

**Comment by sowmyav27** [comment]
**Scale down option is not available on 2.6-head commit id: `b3980c2`**

For cluster members RKE2 clusters
<img width="1423" alt="Screen Shot 2021-08-19 at 10 57 47 PM" src="https://user-images.githubusercontent.com/26032343/130186947-a613af07-8d2f-4f2e-83e9-603a036317d0.png">

**Comment by deniseschannon** [comment]
Cluster members are not expected to be able to scale up or down, so these buttons should be hidden. 
Validated on v2.5.9 that cluster members show the scale up and down buttons, but if you try it, it actually  errors out. 

Verified on v2.6-head b8a44fa. 

Log in as cluster member to view RKE1 and RKE2 clusters. 

RKE1 cluster -  No scale down option available (Correct) 

![Screen Shot 2021-08-23 at 4.04.54 PM.png](https://zube.io/files/rancher/fa581bfe02b9ed52cba22d0cf9da21c0-screen-shot-2021-08-23-at-4-04-54-pm.png)

RKE2 cluster - No scale up/down option (correct), but there is an issue where the nodes are not grouped by machine pools, which is filed as a separate ticket:

![Screen Shot 2021-08-23 at 4.06.47 PM.png](https://zube.io/files/rancher/b1a47e10c5ffed9a3c89b77d7fc3c8cd-screen-shot-2021-08-23-at-4-06-47-pm.png)

This issue should be closed in favor of 
https://github.com/rancher/rancher/issues/34404
